### DISCUSSION TITLE
What's the best way to get search queries on my website to show up in Google?

### DISCUSSION THREAD

**Discussion by florianwalther-private** [vercel/next.js#38588] [Help]
On my website, users can search for resources via a URL query param: `/resources?q=javascript`.
I now want a bunch of these queries to appear in Google search. They don't have to be generated automatically, I'm fine with writing all search terms down by hand.

What's the best way to go about this? Is it enough to add the URLs with the corresponding query param (`q=my-searchterm`) to the Sitemap? Or do I need to create separate page files? O

**Comment by icyJoseph** [comment]
Not an expert here, but I think, if the bot/crawlers, find a link they'll just follow it, and stop only if the target page has set some mechanism to stop them, like `noindex, nofollow` 

So I guess, if you have them on a sitemap file, then they'll be followed by the bot.

However, in some sense, it is much more aggressive than that, see https://github.com/vercel/next.js/discussions/35747, where the bots are trying to index `link`-like text that is part of the `JSON` data used for pages.

So I guess, no separate pages are needed, just links, and the certainty that when a bot arrives there, there'll be content to index. Use the search console to keep track of how this develops over time. It is a bit of a waiting, and patience game.

**Reply by florianwalther-private** [reply]
Thank you. I currently don't have links to these particular searches placed on my website, but maybe it'll work if I create a sitemap that contains them!

**Reply by icyJoseph** [reply]
Nice! Let's see how that goes.

On a perhaps similar note, I think this article is worth reading https://alexskra.com/blog/the-mermaid-is-taking-over-google-search-in-norway/, SEO and Google Bots are a speciality of their own, and I am always amazed at the things true SEO wizards can do.

**Comment by GabenGar** [comment]
Take note for the purpose of SEO you have to slap canonical links everywhere and that means relying less on search params and more on route params.

**Reply by florianwalther-private** [reply]
What do you mean by that?
